sdv/io/local/local.py

```diff
@@ -192,3 +192,90 @@ def write(self, synthetic_data, folder_name, file_name_suffix=None, mode='x'):
+            xl_file = pd.ExcelFile(file_path)
+            sheet_names = xl_file.sheet_names
+
+        for sheet_name in sheet_names:
+            data[sheet_name] = pd.read_excel(
+                file_path,
+                sheet_name=sheet_name,
+                parse_dates=False,
+                decimal=self.decimal,
+                index_col=None
+            )
+
+        return data
+
+    def read(self, file_path, sheet_names=None):
+        """Read data from Excel files and return it along with metadata.
+
+        Args:
+            file_path (str):
+                The path to the Excel file to read.
+            sheet_names (list of str, optional):
+                The names of sheets to read. If None, all sheets are read.
+
+        Returns:
+            tuple:
+                A tuple containing the data as a dictionary and metadata. The dictionary maps
+                table names to pandas DataFrames. The metadata is an object describing the data.
+        """
+        metadata = MultiTableMetadata()
+        if sheet_names is not None and not isinstance(sheet_names, list):
+            raise ValueError("'sheet_names' must be None or a list of strings.")
+
+        data = self._read_excel(file_path, sheet_names)
+        metadata = self._infer_metadata(data)
+        return data, metadata
+
+    def write(self, synthetic_data, file_name, sheet_name_suffix=None, mode='w'):
+        """Write synthetic data to an Excel File.
+
+        Args:
+            synthetic_data (dict):
+                A dictionary mapping table names to pandas DataFrames containing synthetic data.
+            file_name (str):
+                The name of the Excel file to write.
+            sheet_name_suffix (str, optional):
+                A suffix to add to each sheet name.
+            mode (str, optional):
+                The mode of writing to use. Defaults to 'w'.
+                'w': Write sheets to a new file, clearing any existing file that may exist.
+                'a': Append new sheets within the existing file.
+                     Note: You cannot append data to existing sheets.
+        """
+        temp_data = synthetic_data
+        suffix_added = False
+
+        if mode == 'a':
+            temp_data = self._read_excel(file_name)
+            for table_name, table in synthetic_data.items():
+                sheet_name = table_name
+                if sheet_name_suffix:
+                    sheet_name = f'{table_name}{sheet_name_suffix}'
+                    suffix_added = True
+
+                temp_data[sheet_name] = table
+
+        writer = pd.ExcelWriter(file_name)
+        for table_name, table_data in temp_data.items():
+            if sheet_name_suffix and not suffix_added:
+                table_name += sheet_name_suffix
+
+            table_data.to_excel(
+                writer,
+                sheet_name=table_name,
+                float_format=self.float_format,
+                index=False
+            )
```

Comment:
does this overwrite the file if it already existed?

Reply:
This overwrites the sheet. The way excel writer works is that always overwrites the document. There is no 'append' like in `csv`.
That is why in order to 'append' we have to:
* Read the file.
* Extract the sheets that are already in the file.
* Update them.
* Save the file again.

Now that I'm writing this I realize, that I overwrite the sheet name, and I should `append` to it. I'm addressing that in the following commit.

Reply:
I meant if mode is 'w'. The spec says if mode is 'w' and there was anything in the file before, it should be completely overwritten and any other sheets deleted. Is that what happens?

Reply:
Yes, that does happen. It always overwrites even if the file exists.